stripe/src/main/java/com/stripe/android/model/MandateDataParams.kt

```diff
@@ -0,0 +1,97 @@
+
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+/**
+ * Mandate Data parameters for
+ * [confirming a PaymentIntent](https://stripe.com/docs/api/payment_intents/confirm#confirm_payment_intent-mandate_data)
+ * or [confirming a SetupIntent](https://stripe.com/docs/api/setup_intents/confirm#confirm_setup_intent-mandate_data)
+ */
+@Parcelize
+internal data class MandateDataParams internal constructor(
+    private val type: Type = Type.Online,
+    private val typeData: TypeData? = null
+) : StripeParamsModel, Parcelable {
+
+    /**
+     * Create [MandateDataParams] with a [Type] but without [TypeData]
+     */
+    constructor(type: Type = Type.Online) : this(
+        type = type,
+        typeData = null
+    )
+
+    /**
+     * Create [MandateDataParams] with a [Type] and [TypeData]
+     */
+    constructor(typeData: TypeData) : this(
+        type = typeData.type,
+        typeData = typeData
+    )
+
+    override fun toParamMap(): Map<String, Any> {
+        return mapOf(
+            PARAM_CUSTOMER_ACCEPTANCE to mapOf(
+                PARAM_TYPE to type.code
+            ).plus(
+                typeData?.let {
+                    mapOf(it.type.code to it.toParamMap())
+                }.orEmpty()
+            )
+        )
+    }
+
+    /**
+     * The type of customer acceptance information included with the Mandate.
+     */
+    enum class Type(internal val code: String) {
+        Online("online")
+    }
+
+    sealed class TypeData(
+        internal val type: Type
+    ) : StripeParamsModel, Parcelable {
+
+        /**
+         * If this is a Mandate accepted online, this hash contains details about the online acceptance.
+         *
+         * [mandate_data.customer_acceptance.online](https://stripe.com/docs/api/payment_intents/confirm#confirm_payment_intent-mandate_data-customer_acceptance-online)
+         */
+        @Parcelize
+        data class Online(
+            /**
+             * The IP address from which the Mandate was accepted by the customer.
+             *
+             * [mandate_data.customer_acceptance.online.ip_address](https://stripe.com/docs/api/payment_intents/confirm#confirm_payment_intent-mandate_data-customer_acceptance-online-ip_address)
+             */
+            private val ipAddress: String,
+
+            /**
+             * The user agent of the browser from which the Mandate was accepted by the customer.
+             *
+             * [mandate_data.customer_acceptance.online.user_agent](https://stripe.com/docs/api/payment_intents/confirm#confirm_payment_intent-mandate_data-customer_acceptance-online-user_agent)
+             */
+            private val userAgent: String
+        ) : TypeData(Type.Online) {
+            override fun toParamMap(): Map<String, Any> {
+                return mapOf(
+                    PARAM_IP_ADDRESS to ipAddress,
+                    PARAM_USER_AGENT to userAgent,
+                    PARAM_INFER_FROM_CLIENT to true
```

Comment:
so is this just temporary until you make this public? Then presumably it shouldn't always map infer_from_client to true

Reply:
Good catch, fixed.